plugins/Cardinal/src/Ildaeil.cpp

```diff
@@ -1031,6 +1044,20 @@ struct IldaeilWidget : ImGuiWidget, IdleCallback, Thread {
             fileBrowserHandle = nullptr;
         }
 
+        if (fPluginGenericUI != nullptr && module->fCarlaHostHandle != nullptr) {
+            bool expanderPresent = (module->leftExpander.module && module->leftExpander.module->model == modelIldaeilExpIn8);
+            float *messagesFromExpander = (float*)module->leftExpander.consumerMessage;// could be invalid pointer when !expanderPresent, so read it only when expanderPresent
+            if (expanderPresent) {
+                for (uint i = 0; i < 8; i++) {
+                    if (i < fPluginGenericUI->parameterCount && module->leftExpander.module->inputs[i].isConnected()) {
+                        float scaled_param = (messagesFromExpander[i] + 10.0) * (fPluginGenericUI->parameters[i].max - fPluginGenericUI->parameters[i].min) / (20.0 + fPluginGenericUI->parameters[i].min);
+                        fPluginGenericUI->values[i] = scaled_param;
+                        carla_set_parameter_value(module->fCarlaHostHandle, 0, 0, scaled_param);
```

Comment:
@falkTX I'm unsure about that part, if there's a more elegant way. How does that look to you? From my test and understanding of this module, both values and call to `carla_set_parameter_value` have to be made in order to update both the DSP and the UI.

Reply:
yeah this should be done in a different way, I will handle it soon